hibernate-core/src/main/java/org/hibernate/boot/models/xml/internal/ManagedTypeProcessor.java

```diff
@@ -234,6 +234,12 @@ private static void processEntityMetadata(
 					memberAdjuster,
 					xmlDocumentContext
 			);
+			AttributeProcessor.processVersionAttribute(
+					attributes.getVersion(),
+					classDetails,
+					classAccessType,
+					xmlDocumentContext
+			);
```

Comment:
Couldn't a `<version>` attribute also be defined in a `<mapped-superclass>`? I think I'd rather put version processing within `AttributeProcessor.processAttributes`, since it can be found in any `<attributes>` container, wdyt?